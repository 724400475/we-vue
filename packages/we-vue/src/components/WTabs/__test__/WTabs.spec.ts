import { shallowMount, mount } from '@vue/test-utils'
import Tabs from '../WTabs'
import Tab from '../../WTab'

describe('tabs', () => {
  test('create', () => {
    const wrapper = shallowMount(Tabs, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-tabs')
    expect(wrapper.classes()).toContain('wv-tabs')
  })

  test('create with tab items', () => {
    const wrapper = mount(Tabs, {
      attachToDocument: true,
      propsData: {},
      slots: {
        default: [Tab, Tab, Tab],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    expect(wrapper.name()).toBe('w-tabs')
    expect(wrapper.classes()).toContain('wv-tabs')
  })

  test('click tab items', () => {
    const wrapper = mount(Tabs, {
      attachToDocument: true,
      propsData: {},
      slots: {
        default: [Tab, Tab, Tab],
      },
    })

    wrapper
      .findAll(Tab)
      .at(1)
      .trigger('click')
  })
})

describe('tab', () => {
  test('create', () => {
    const wrapper = mount(Tabs, {
      attachToDocument: true,
      propsData: {},
      slots: {
        default: [Tab, Tab, Tab],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('destroy', () => {
    const wrapper = mount(Tabs, {
      attachToDocument: true,
      propsData: {},
      slots: {
        default: [Tab, Tab, Tab],
      },
    })

    wrapper.findAll(Tab).at(0).vm.$destroy()

    expect(wrapper.vm.tabs).toHaveLength(2)
  })
})
