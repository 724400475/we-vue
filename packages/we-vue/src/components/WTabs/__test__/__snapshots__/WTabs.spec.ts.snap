// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab create 1`] = `
<div class="wv-tabs wv-tabs--line">
  <div class="wv-tabs__wrap wv-hairline--top-bottom">
    <div class="wv-tabs__nav wv-tabs__nav--line">
      <div class="wv-tabs__line"></div>
    </div>
  </div>
  <div class="wv-tabs__content">
    <div class="wv-tab__pane"></div>
    <div class="wv-tab__pane"></div>
    <div class="wv-tab__pane"></div>
  </div>
</div>
`;

exports[`tabs create with tab items 1`] = `
<div class="wv-tabs wv-tabs--line">
  <div class="wv-tabs__wrap wv-hairline--top-bottom">
    <div class="wv-tabs__nav wv-tabs__nav--line">
      <div class="wv-tabs__line"></div>
      <div class="wv-tab wv-tab--active"><span class="wv-ellipsis"></span></div>
      <div class="wv-tab"><span class="wv-ellipsis"></span></div>
      <div class="wv-tab"><span class="wv-ellipsis"></span></div>
    </div>
  </div>
  <div class="wv-tabs__content">
    <div class="wv-tab__pane"></div>
    <div class="wv-tab__pane" style="display: none;"></div>
    <div class="wv-tab__pane" style="display: none;"></div>
  </div>
</div>
`;
